### ISSUE TITLE
GetVariantDate returns wrong data on 32-bit platform

### ISSUE THREAD

**Issue by lkarlslund** [go-ole/go-ole#177]
The call to GetVariantDate passes the time as a float64, which is in turn converted to an uintptr. This works as expected on a 64-bit platform, but on 32-bit the value is truncated, and wrong results are returned.

I've fixed this in my own fork, but I have no clue how to submit a pull request that does not completely destroy the module paths in this original version. So rather old-school, here is my fix:

com.go:
```
const sixtyfour = uint64(^uint(0)) == ^uint64(0)

// GetVariantDate converts COM Variant Time value to Go time.Time.
func GetVariantDate(value uint64) (time.Time, error) {
	var st syscall.Systemtime
	var r uintptr
	if !sixtyfour {
		// 32-bit
		lo := uintptr(value)
		hi := uintptr(value >> 32)
		r, _, _ = procVariantTimeToSystemTime.Call(uintptr(lo), uintptr(hi), uintptr(unsafe.Pointer(&st)))
	} else {
		r, _, _ = procVariantTimeToSystemTime.Call(uintptr(value), uintptr(unsafe.Pointer(&st)))
	}

	if r != 0 {
		return time.Date(int(st.Year), time.Month(st.Month), int(st.Day), int(st.Hour), int(st.Minute), int(st.Second), int(st.Milliseconds/1000), time.UTC), nil
	}
	return time.Time{}, errors.New("Could not convert to time, passing current time.")
}
```

variant.go:
```
	case VT_DATE:
		// VT_DATE type will either return time.Time, either correct one or uninitialied one.
		d := uint64(v.Val)
		date, _ := GetVariantDate(d)
		return date
```

Hope this helps someone else.

**Comment by mattn** [comment]
Could you please send me pull-request?

**Comment by konimarti** [comment]
@mattn @lkarlslund  I think there is already a pull request that addresses this issue: #155

**Comment by mattn** [comment]
merged #155
